docs/design/structured-concurrency.md

```diff
@@ -0,0 +1,63 @@
+* **HttpCall** - A single HTTP request/response pair. 
+* **HttpClientEngine** - A component responsible for executing an HTTP request and returning an HttpCall
+* **ExecutionContext** - Operation scoped (mutable) context used to drive an operation to completion
+
+
+## Scopes
+
+The SDK has three implementations of `CoroutineScope`:
+
+1. [ExecutionContext](https://github.com/awslabs/smithy-kotlin/blob/main/runtime/runtime-core/common/src/aws/smithy/kotlin/runtime/operation/ExecutionContext.kt#L16)
+2. [HttpClientEngine](https://github.com/awslabs/smithy-kotlin/blob/main/runtime/protocol/http-client/common/src/aws/smithy/kotlin/runtime/http/engine/HttpClientEngine.kt#L20)
+3. [HttpCall](https://github.com/awslabs/smithy-kotlin/blob/main/runtime/protocol/http/common/src/aws/smithy/kotlin/runtime/http/response/HttpCall.kt#L28)
+    
+
+`ExecutionContext` implements `CoroutineScope` to provide a place for any background work to be done as part of implementing an operation. 
+Its scope begins and ends with an operation. The only place it is utilized is to launch background task(s) to process input event streams 
+(pull data from the customer input event stream, transform it, and write it to the HTTP body). 
+
+`HttpClientEngine` implements `CoroutineScope` and is used to launch HTTP requests. The scope of the engine is from creation to when it is closed.
+Making HTTP request tasks children of this scope means that an engine won’t be shutdown until in-flight requests are complete.
+The engine scope is a [SupervisorJob](https://kotlinlang.org/api/kotlinx.coroutines/kotlinx-coroutines-core/kotlinx.coroutines/-supervisor-job.html) such that individual HTTP requests can fail independent of one another without causing all requests to be cancelled.
+
+`HttpCall` implements `CoroutineScope` to provide a place for any background work required to fulfill an HTTP request/response. It is described
+in more detail in the following section.
+
+### HttpCall Scope
+
+Individual HTTP calls get their own `CoroutineContext` whose scope begins with executing the request and ends when `HttpCall::complete()` is invoked.
+This context/scope is referred to as the "call context". The call context is tied to the `HttpEngine` and is used primarily to launch background work
+required to fulfill executing the request and response. It is the parent to the `HttpClientEngine::roundTrip` invocation as well as any work required
+to send the request body or fulfill the response body (e.g. reading off the wire and adapting to the the SDK I/O primitives). 
+
+Visually the parent/child relationships look like this:
+
+![HttpCall Context](resources/HttpCall-Context.png)
+
+
+The call context is _not_ a child of the operation context it is invoked from. It is however configured (manually) to be cancelled if the caller cancels the operation (dotted red line).
+
+### FAQ
+
+**Why is the call context not a child of the operation context**
```

Comment:
**Nit**: Add question mark (`?`) at the end

Reply:
🤦